src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -857,6 +847,42 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
             ProjectImportElement importAfter = traversalProject.CreateImportElement(@"$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\SolutionFile\ImportAfter\*");
             importAfter.Condition = @"'$(ImportByWildcardBeforeSolution)' != 'false' and exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\SolutionFile\ImportAfter')"; // Avoids wildcard perf problem
 
+            ProjectPropertyGroupElement directorySolutionPropsPropertyGroup = traversalProject.CreatePropertyGroupElement();
```

Comment:
Can extract to method and pass in the traversal project objects that you add stuff to.

Reply:
The problem is I'm making two PropertyGroups but I can't add their children until I've added the PropertyGroup items to the document.